smartmin/management/__init__.py

```diff
@@ -7,24 +7,24 @@
 from django.contrib.contenttypes.models import ContentType
 from django.contrib.auth.models import Permission, Group
 from django.core.exceptions import ObjectDoesNotExist
-from django.db.models.signals import post_syncdb
+from django.db.models.signals import post_migrate
 from guardian.management import create_anonymous_user
 from guardian.shortcuts import assign_perm, remove_perm
 from guardian.utils import get_anonymous_user
 
 
-def is_last_model(kwargs):
+def is_last_model(app_config):
     """
-    Returns whether this is the last post_syncdb called in the application.
+    Returns whether this is the last post_migrate called in the application.
     """
     # If the application specifies it, use their permissions app. This app should be
     # the last app needs to be the last app in INSTALLED_APPS which uses smartmin permissions.
     permissions_app = getattr(settings, 'PERMISSIONS_APP', None)
     if permissions_app:
-        return kwargs['app'].__name__ == "%s.models" % permissions_app
+        return app_config.name == "%s.models" % permissions_app
 
     # Otherwise, run it for each of the last five apps in INSTALLED_APPS
-    return kwargs['app'].__name__ in ["%s.models" % app for app in settings.INSTALLED_APPS[-10:]]
+    return app_config.name in settings.INSTALLED_APPS[-10:]
```

Comment:
Oh and also checked that Django versions 1.7+ return the app configs in the same order as they're migrated.